tests/cases/compiler/emitClassExpressionInDeclarationFile.ts

```diff
@@ -0,0 +1,30 @@
+// @declaration: true
+export var simpleExample = class {
+    static getTags() { }
+    tags() { }
+}
+export var circularReference = class C {
+    static getTags(c: C): C { return c }
+    tags(c: C): C { return c }
+}
+
+// repro from #15066
+export class FooItem {
+    foo(): void { }
+    name?: string;
+}
+
+export type Constructor<T> = new(...args: any[]) => T;
+export function WithTags<T extends Constructor<FooItem>>(Base: T) {
+    return class extends Base {
+        static getTags(): void { }
+        tags(): void { }
+    }
+}
+
+export class Test extends WithTags(FooItem) {}
+
```

Comment:
I tried adding an ad-hoc check, but I couldn't get it to distinguish between this case, where the class expression extends a class declaration, and the mixin case, where the class expression extends a parameter. There are two problems with the latter case:

1. The mixed-in class could easily be a non-exported class declaration.
2. Class declarations that call the mix-in are no longer enclosed in the mix-in function, making it difficult to track back to the original class expression's context in the displayType machinery.

This export will not cause errors in any case, because the class expression is exported as a type literal, so `Base` isn't actually referenced &mdash; just its contents.

tl;dr &mdash; I don't think it's worthwhile to issue an error here.